PEUQSE/InverseProblem.py

```diff
@@ -1885,6 +1885,137 @@ def exportPostPermutationStatistics(self, searchType=''): #if it is an mcmc run,
+        elif continueSampling == True:
+            walkerStartPoints = self.mcmc_last_point_sampled 
+        emcee_sampler = emcee.EnsembleSampler(self.mcmc_nwalkers, numParameters, log_prob_fn=self.getLogP)    
+        for trialN in range(0,1000):#Todo: This number of this range is hardcoded but should probably be a user selection.
+            try:
+                emcee_sampler.run_mcmc(walkerStartPoints, nEnsembleSteps)
+                break
+            except Exception as exceptionObject:
+                #TODO: come up with exception summaries like this.
+                if "finite" in str(exceptionObject): #This means there is an error message from zeus saying " Invalid walker initial positions!  Initialise walkers from positions of finite log probability."
+                    print("One of the starting points has a non-finite probability. Picking new starting points. If you see this message like an infinite loop, consider trying the doEnsembleSliceSampling optional argument of walkerInitialDistributionSpread. It has a default value of 1.0. Reducing this value to 0.25, for example, may work if your initial guess is near the maximum of the posterior distribution.")
+                    #Need to make the sampler again, in this case, to throw away anything that has happened so far
+                    walkerStartPoints = self.generateInitialPoints(initialPointsDistributionType=walkerInitialDistribution, numStartPoints = self.mcmc_nwalkers, relativeInitialDistributionSpread=walkerInitialDistributionSpread) 
+                    emcee_sampler = emcee.EnsembleSampler(self.mcmc_nwalkers, numParameters, log_prob_fn=self.getLogP)    
+                elif "maxiter" in str(exceptionObject): #This means there is an error message from zeus that the max iterations have been reached.
+                    print("WARNING: One or more of the Ensemble Slice Sampling walkers encountered an error. The value of mcmc_maxiter is currently", mcmc_maxiter, "you should increase it, perhaps by a factor of 1E2.")
+                else:
+                    print(str(exceptionObject))
+                    sys.exit()
+        #Now to keep the results:
+        #TODO: when implementing convergence diagnostics, extract more chain information from get_chain without flattening (i think)
+        adjusted_mcmc_burn_in_length = int(self.mcmc_burn_in_length / self.mcmc_nwalkers)
+        self.post_burn_in_samples = emcee_sampler.get_chain(flat=True, discard = adjusted_mcmc_burn_in_length )
+        self.post_burn_in_log_posteriors_un_normed_vec = np.atleast_2d(emcee_sampler.get_log_prob(flat=True, discard=adjusted_mcmc_burn_in_length)).transpose() #Needed to make it 2D and transpose.
+        self.mcmc_last_point_sampled=emcee_sampler.get_last_sample() #gets the actual last sample after all chains have been flattened to one.
```

Comment:
Continue sampling also needs to be made for the emcee sampler. Right now, it does not keep the last point of all chains (which would be a vector).  Right now it keeps a scalar, but we need the vector.
https://github.com/AdityaSavara/PEUQSE/issues/217